test/contrib/azureblob_test.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+Integration tests for azureblob module.
+"""
+
+import os
+import unittest
+
+import luigi
+from luigi.contrib.azureblob import AzureBlobClient, AzureBlobTarget
+
+client = AzureBlobClient(os.environ.get("ACCOUNT_NAME"), os.environ.get("ACCOUNT_KEY"), os.environ.get("SAS_TOKEN"))
```

Comment:
Is there anyway to mock this for Travis testing?

Reply:
Not to my knowledge. Azure Storage does come with an emulator but [unfortunately it only works on windows](https://docs.microsoft.com/en-us/azure/storage/common/storage-use-emulator)

There is an open source clone, but I was not very keen on adding yet another dependency to such a small test. I can DM you an account name/key if you want to try out these tests

Reply:
There look to be some some way to test this stuff without real accounts: https://github.com/Azure/azure-storage-python/tree/master/tests

And this is from the python SDK already being used by your additions.

Reply:
I am going to look into if it is possible to mock or create some kind of stubs. From the looks of the azure-storage-python tests, it looks like it uses some proxy, but I see no mocking.

I am going to try out https://github.com/azure/azurite but it would just end up introducing yet another dependency.

Reply:
My main concern here is accepting a module which cannot be easily tested. It's one thing if it just can't be ran on Travis. But another if it can't be ran locally with tox. Of course, without an Azure account configured for these tests.

Just looking to minimize future maintenance burden :)

Reply:
Would making this addition to tox.ini work?

```
[testenv:azureblob]
passenv = ACCOUNT_NAME ACCOUNT_KEY SAS_TOKEN
deps =
  azure-storage
commands =
  python -m unittest -v test.contrib.azureblob_test
```

and append `azureblob` in 

```
[tox]
envlist = ..... azureblob
```

then run these tests with

```bash
tox -e azureblob
```

Pardon my inexperience. Python is not my primary language for bread and butter :)

Reply:
@dlstadther Would the above change in tox work?

Reply:
It _would_ work, but that departs from our current test state. I'd really like to avoid requiring the use of actual credentials for testing

@Tarrasch Do you have any suggestions on best route to approve given the issues around testing without actual Azure credentials? I'd like not to keep blocking this PR, but I also wish to avoid maintaining code which requires manual setup to test.

Reply:
@manish @dlstadther Re: "I'd really like to avoid requiring the use of actual credentials for testing"
Once the azure storage emulator is running(either [Azurite](https://github.com/Azure/Azurite) for linux or [emulator](https://docs.microsoft.com/en-us/azure/storage/common/storage-use-emulator) for windows), you just have to use the azure blob client like so:
```python
from azure.storage.blob import BlockBlobService

blob_client = BlockBlobService(is_emulated=True)  # will automatically use the default local connection string and keys
```
For both azurite and the windows emulator, the blob service runs on a default port and with a default key, which is built into the azure storage python client library(see [docs](https://azure-storage.readthedocs.io/ref/azure.storage.blob.blockblobservice.html)).

For an example of how to use azurite in travis, this is what I do [(here)](https://github.com/shikharsg/zarr/blob/abs_store/.travis.yml):
```
before_install:
  - docker pull arafato/azurite
  - mkdir ~/blob_emulator
  - docker run -e executable=blob -d -t -p 10000:10000 -v ~/blob_emulator:/opt/azurite/folder arafato/azurite
```

Reply:
I made the changes as you suggested, but I made it in tox so that I can run it on multiple python versions. Thanks a lot for your help @shikharsg